examples/pytorch/llm/utils/utils.py

```diff
@@ -0,0 +1,165 @@
+import os
+from typing import List, Optional, Tuple
+
+import matplotlib.pyplot as plt
+import torch
+import torch.distributed as dist
+from torch import dtype as Dtype
+from torch.nn import Module
+from transformers import GenerationConfig, TextStreamer
+
+from swift import get_logger
+from swift.utils.tb_utils import (TB_COLOR, TB_COLOR_SMOOTH,
+                                  read_tensorboard_file, tensorboard_smoothing)
+
+os.environ['TOKENIZERS_PARALLELISM'] = 'true'
+logger = get_logger()
+
+DEFAULT_PROMPT = """Here's a conversation between a human and an AI assistant. \
+The AI assistant provides detailed, friendly answers for the human.
+
+### Human:
+{instruction}
+
+### AI:
+"""
+
+DTYPE_MAPPING = {
+    'fp16': torch.float16,
+    'bf16': torch.bfloat16,
+    'fp32': torch.float32
+}
+
+
+def get_dist_setting() -> Tuple[int, int, int]:
+    """return rank, local_rank, world_size"""
+    rank = int(os.getenv('RANK', -1))
+    local_rank = int(os.getenv('LOCAL_RANK', -1))
+    world_size = int(os.getenv('WORLD_SIZE', 1))
+    return rank, local_rank, world_size
+
+
+def is_dist():
+    rank = int(os.getenv('RANK', -1))
```

Comment:
When does the variable RANK exist?

Reply:
torchrun will sets environment variables